vllm/v1/core/sched/scheduler.py

```diff
@@ -1335,7 +1340,8 @@ def _connector_finished(
             assert len(self.kv_cache_config.kv_cache_groups) == 1
             return self.connector.request_finished(request, block_ids[0])
         else:
-            return self.connector.request_finished(request, block_ids)
+            assert isinstance(self.connector, SupportsHMA)
```

Comment:
I think we should just change the `if not supports_hma(self.connector)` above to `if not isinstance(self.connector, SupportsHMA)`.

Also the `else` isn't needed